examples/YOLOv8-SAHI-Inference-Video/YOLOv8_SAHI_ObjectTracking.py

```diff
@@ -0,0 +1,193 @@
+                boxes_np = np.array(boxes_list, dtype=np.int16)[:, :-1]
+                trackers = tracker.update(boxes_np, conf, frame)
+                for track in trackers:
+                    bbox = track[0:4]  # Get x1, y1, x2, y2 coordinates directly
+                    track_id = track[4]
+
+                    data.append({"id": int(track_id), "bbox": np.array(bbox, dtype=np.int16)})
+
+                    cvzone.cornerRect(
+                        frame,
+                        (int(bbox[0]), int(bbox[1]), int(bbox[2]), int(bbox[3])),
+                        l=0,
+                        rt=1,
+                        t=1,
+                        colorR=(0, 0, 255),
+                        colorC=(0, 0, 255),
+                    )
+                    cvzone.putTextRect(
+                        frame,
+                        f"Track ID: {int(track_id)}",
+                        (max(0, int(bbox[0])), max(0, int(bbox[1]))),
+                        1,
+                        2,
+                        colorT=(255, 255, 255),
+                        colorR=(0, 0, 255),
+                    )
+
+        if without_tracking:
+            boxes_list = []
+            clss_list = []
+            for ind, _ in enumerate(object_prediction_list):
+                boxes = (
+                    object_prediction_list[ind].bbox.minx,
+                    object_prediction_list[ind].bbox.miny,
+                    object_prediction_list[ind].bbox.maxx,
+                    object_prediction_list[ind].bbox.maxy,
+                )
+                clss = object_prediction_list[ind].category.name
+                boxes_list.append(boxes)
+                clss_list.append(clss)
+
+            for box, cls in zip(boxes_list, clss_list):
+                x1, y1, x2, y2 = box
+                cv2.rectangle(frame, (int(x1), int(y1)), (int(x2), int(y2)), (56, 56, 255), 2)
+                label = str(cls)
+                t_size = cv2.getTextSize(label, 0, fontScale=0.6, thickness=1)[0]
+                cv2.rectangle(
+                    frame, (int(x1), int(y1) - t_size[1] - 3), (int(x1) + t_size[0], int(y1) + 3), (56, 56, 255), -1
+                )
+                cv2.putText(
+                    frame, label, (int(x1), int(y1) - 2), 0, 0.6, [255, 255, 255], thickness=1, lineType=cv2.LINE_AA
+                )
+
+        if view_img:
+            if is_webcam:
+                cv2.imshow("YOLOv8", frame)
+            else:
+                cv2.imshow(Path(source).stem, frame)
+        if save_img:
+            video_writer.write(frame)
+
+        if cv2.waitKey(1) & 0xFF == ord("q"):
+            break
+    video_writer.release()
+    videocapture.release()
+    cv2.destroyAllWindows()
+
+
+def parse_opt():
+    """Parse command line arguments."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--weights", type=str, default="yolov8n.pt", help="initial weights path")

```

Comment:
@MuhammadBilal848 would be awesome if you could include some args as `zoom_factor`, `prediction_threshold` etc

Reply:
I added zoom_factor and conf_threshold. albeit zoom_factor is not being utilized but conf_threshold is.

Reply:
can you give more detail on the issue about zoom_factor? @MuhammadBilal848